File: java_version/double_pointer/window_sum.java
package double_pointer;

public class window_sum {
    public int[] winSum(int[] nums, int k) {

        /*
        For array [1,2,7,8,5], moving window size k = 3.
        1 + 2 + 7 = 10
        2 + 7 + 8 = 17
        7 + 8 + 5 = 20
        return [10,17,20]
         */
        if(k > nums.length || nums.length == 0){
            return new int[0];
        }
        int[] result = new int[nums.length - k + 1];
        int left = 0, right = k - 1, sum = 0;
        for(int i = left; i <= right; i++){
            sum += nums[i];
        }
        result[left] = sum;
        while (left + 1 + k <= nums.length){
            sum = sum - nums[left];
            right += 1;
            sum = sum + nums[right];
            left += 1;
            result[left] = sum;
        }
        return result;
    }
}

File: python_version/pointer_array/window_sum.py
class Solution:
    """
    @param nums: a list of integers.
    @param k: length of window.
    @return: the sum of the element inside the window at each moving.
    """

    def winSum(self, nums, k):
        # write your code here
        if nums == [] or nums is None or k == 0 or len(nums) < k:
            return []
        result, sum_p = [], 0
        for i in range(k):
            sum_p += nums[i]
        result.append(sum_p)
        move_space = len(nums) - k + 1
        i, left, right = 1, 0, k

        while i < move_space:
            left_value = nums[left]
            right_value = nums[right]
            sum_p -= left_value
            sum_p += right_value

            result.append(sum_p)

            left += 1
            right += 1
            i += 1
        return result
